cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/MockCruiseControl.java

```diff
@@ -0,0 +1,158 @@
+
+        // Rebalance response with no goals set - non-verbose
+        String json = getJsonFromResource("CC-Rebalance-no-goals.json");
+
+        ccServer
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.JSON.key, "true"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.DRY_RUN.key, "true|false"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.VERBOSE.key, "false"))
+                                .withPath(CruiseControlEndpoints.REBALANCE.path))
+                .respond(
+                        response()
+                                .withBody(json)
+                                .withHeaders(header("User-Task-ID", "rebalance-no-goals")));
+
+        // Rebalance response with no goals set - verbose
+        String jsonVerbose = getJsonFromResource("CC-Rebalance-no-goals-verbose.json");
+
+        ccServer
+                .when(
+                        request()
+                                .withMethod("POST")
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.JSON.key, "true"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.DRY_RUN.key, "true|false"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.VERBOSE.key, "true"))
+                                .withPath(CruiseControlEndpoints.REBALANCE.path))
+                .respond(
+                        response()
+                                .withBody(jsonVerbose)
+                                .withHeaders(header("User-Task-ID", "rebalance-no-goals-verbose")));
+
+    }
+
+    private static void setupCCUserTasksResponse(ClientAndServer ccServer) throws IOException, URISyntaxException {
+
+        // User tasks response for the rebalance request with no goals set (non-verbose)
+        String json = getJsonFromResource("CC-User-task-rebalance-no-goals.json");
+
+        ccServer
+                .when(
+                        request()
+                                .withMethod("GET")
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.JSON.key, "true"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.FETCH_COMPLETE.key, "true"))
+                                .withQueryStringParameter(Parameter.param(CruiseControlParameters.USER_TASK_IDS.key, "rebalance-no-goals"))
+                                .withPath(CruiseControlEndpoints.USER_TASKS.path))
+                .respond(
+                        response()
+                                .withBody(json)
+                                .withHeaders(header("User-Task-ID", "user-task-rebalance-no-goals")));
```

Comment:
Would it be worth setting Strings like "user-task-rebalance-no-goals" as variable fields of MockCruiseControl as well?

Reply:
Yes it would because that would make writing tests easier and later we may add quite a lot of these to cover difference scenarios.